### PR TITLE
feat(graph): add target groups and technology icon

### PR DESCRIPTION
**Pull request by xiongemi**
## Expected Behavior

Screenshot 2024-04-23 at 1 42 14 PM
Screenshot 2024-04-23 at 1 42 08 PM
Screenshot 2024-04-23 at 1 42 02 PM

## Related Issue(s)

Fixes #

### CODE REVIEW

File: graph/ui-icons/src/lib/framework-icons.stories.tsx [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+import { Framework, frameworkIcons } from './framework-icons';
+
+const meta: Meta<typeof frameworkIcons> = {
+  component: () => (
+    <>
+      {Object.keys(frameworkIcons).map((key) => (
+        <>
+          <div>{key}</div>
+          <div className="h-10 w-10">
+            {frameworkIcons[key as Framework].image}
```
Comment (FrozenPandaz) [inline graph/ui-icons/src/lib/framework-icons.stories.tsx:11]: Please handle when the `frameworkIcons` does not have the technology.

File: graph/ui-icons/src/lib/technology-icon.tsx [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+  technology?: string;
+  showTooltip?: boolean;
+}
+
+export function TechnologyIcon({
+  technology,
+  showTooltip,
+}: TechnologyIconProps) {
+  if (!technology || !frameworkIcons[technology as Framework]) {
+    return null;
```
Comment (FrozenPandaz) [inline graph/ui-icons/src/lib/technology-icon.tsx:13]: We're not showing anything at the moment?

Can we please update this to show a generic element with the first character of the technology please?

File: packages/gradle/src/plugin/nodes.ts [hunk 3]
```diff
@@ -139,13 +134,17 @@ export const createNodes: CreateNodes<GradlePluginOptions> = [
       calculatedTargets[hash] = {
         name: projectName,
         targets,
-        targetGroups,
+        metadata: {
+          targetGroups,
+          technologies: ['gradle'],
```
Comment (FrozenPandaz) [inline packages/gradle/src/plugin/nodes.ts:139]: Remove this from the project, only have it on the targets.

File: packages/gradle/src/plugin/nodes.ts [hunk 4]
```diff
@@ -139,13 +134,17 @@ export const createNodes: CreateNodes<GradlePluginOptions> = [
       calculatedTargets[hash] = {
```
Comment (FrozenPandaz) [inline packages/gradle/src/plugin/nodes.ts:134]: Can this line be combined with the above line so we don't have to keep them in sync?

```
calculatedTargets[hash] = project
```

File: nx.json [hunk 5]
```diff
@@ -164,7 +179,10 @@
         "{projectRoot}/.storybook/**/*",
         "{projectRoot}/tsconfig.storybook.json"
       ],
-      "cache": true
+      "cache": true,
+      "metadata": {
+        "technologies": ["storybook"]
+      }
```
Comment (FrozenPandaz) [inline nx.json:185]: Can you remove these? We'll add these separately to the plugins

**Review by FrozenPandaz — CHANGES_REQUESTED** [review]
@MaxKless Could you take a look at the UI code please?

File: graph/ui-icons/src/lib/technology-icon.tsx (hunk 2 above)
Comment (xiongemi) [reply graph/ui-icons/src/lib/technology-icon.tsx:13]: i updated the ui to be something like:
Screenshot 2024-04-25 at 11 05 48 AM
Screenshot 2024-04-25 at 11 05 42 AM

File: graph/ui-icons/src/lib/framework-icons.stories.tsx (hunk 1 above)
Comment (xiongemi) [reply graph/ui-icons/src/lib/framework-icons.stories.tsx:11]: done

File: nx.json (hunk 5 above)
Comment (xiongemi) [reply nx.json]: removed

File: packages/gradle/src/plugin/nodes.ts (hunk 4 above)
Comment (xiongemi) [reply packages/gradle/src/plugin/nodes.ts]: changed

File: packages/gradle/src/plugin/nodes.ts (hunk 3 above)
Comment (xiongemi) [reply packages/gradle/src/plugin/nodes.ts]: removed

**Review by FrozenPandaz — APPROVED** [review]

**Pull request merged**